pkg/rcmanager/node.go

```diff
@@ -0,0 +1,206 @@
+package rcmanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	networkingv1 "github.com/oecp/rama/pkg/apis/networking/v1"
+	"github.com/oecp/rama/pkg/constants"
+	"github.com/oecp/rama/pkg/utils"
+	apiv1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	runtimeutil "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+)
+
+const ReconcileNode = "ReconcileNode"
+
+// Full update. Update remote vtep expect status
+func (m *Manager) reconcileNode() error {
+	klog.Infof("[RemoteCluster] Starting reconcile node from cluster %v", m.ClusterName)
+	nodes, err := m.nodeLister.List(labels.NewSelector())
+	if err != nil {
+		return err
+	}
+	vteps, err := m.remoteVtepLister.List(utils.SelectorClusterName(m.ClusterName))
+	if err != nil {
+		return err
+	}
+
+	add, update, remove := m.diffNodeAndVtep(nodes, vteps)
+	var (
+		wg  sync.WaitGroup
+		cur = metav1.Now()
+	)
+	wg.Add(3)
```

Comment:
https://stackoverflow.com/questions/65213707/where-to-put-wg-add

Reply:
Well, it's a mistake to use it in a go routine. 

But actually what my point is, `wg.Add(3)` is not friendly for later maintenance, because it's always hard for someone to count how many goroutine we have in the codes, which might bring problems if a goroutine is actually added but without increase the number of `wg.Add(xxx)`.

Maybe write `wg.Add(1)` for every separate goroutine (like three times for three gorouine) just before the `go func() {}` is better.